### PR TITLE
I #17: CI and CD Pipeline initial

### LINKED ISSUE #17: Create CI/CD pipeline
Github actions to execute the tests, unit, integration, end-to-end. We can improve the pipeline in future as the scope and codebase grows

### What CI/CD pipeline will do?
1. Lint code
2. Lint terraform
3. Unit tests and measure code coverage
4. Integration test (future)
5. Security check (future)
6. Build/package libs/docker image if necessary

### Task
1. Make a github action
2. Use separate steps for linting and testing etc.
3. For dev branches it will only check lint, tests and security
4. For master branch, it will provide a report if the master is deployable
5. If code is pushed to master, it will deploy to dev branch
6. For any git release tag, it will deploy to prod depending on tag

### PR DESCRIPTION
**Pull request by melon-ruet**
Fixes #17

Resolves #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [x] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

CI/CD with linting and testing and local developemnt process

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [x] I have added or updated necessary documentation to reflect the changes.
- [x] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [x] I have checked for any potential conflicts with other branches or pull requests.
- [x] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

Please ignore improvement

## Known Issues (optional)

Dev environment is not ready with localstack

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: .github/workflows/deploy-branch.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
+name: "Branch Deployment"
```
Comment (melon-ruet) [inline .github/workflows/deploy-branch.yml:1]: Note that this is in progress and a separate ticket will be created

File: .github/workflows/destroy-branch.yml
Comment (melon-ruet) [inline .github/workflows/destroy-branch.yml:1]: Note that this is in progress and a separate ticket will be created

File: .github/workflows/deploy-branch.yml
Comment (melon-ruet) [inline .github/workflows/deploy-branch.yml:1]: Note that this is in progress and a separate ticket will be created

File: README.md [hunk 2]
```diff
@@ -2,4 +2,15 @@
 BloodConnect
 
 ## Docs
-- [Home](https://craftsmenltd.github.io/BloodConnect)
\ No newline at end of file
+- [Home](https://craftsmenltd.github.io/BloodConnect)
+
+<br>
+<br>
+For mac os Docker
```
Comment (sakibstark11) [inline README.md:9]: what happened here?
I thought i wrote a comprehensive guide?

Comment (sakibstark11) [reply README.md:9]: I think you wanted to remove these lines from here?

**Review by sakibstark11 — DISMISSED** [review]
nice work pal

**Pull request merged**